packages/help-center/src/happychat-window-communicator.ts

```diff
@@ -4,69 +4,71 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from '@wordpress/element';
 import { useQueryClient } from 'react-query';
 import { useSelector } from 'react-redux';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 /**
  * Internal Dependencies
  */
 import { HELP_CENTER_STORE, SITE_STORE } from './stores';
-import type { WindowState } from './types';
 
-/**
- * This hook attaches an event listener to the window to listen to Happychat messages that com from https://widgets.wp.com/calypso-happychat/
- * It collects information from the help-center store and forwards it to Happychat. It also forwards the unread count and the state changes (ended, blurred, closed, opened) to whoever uses the hook
- *
- * @param onStateChange a callback that will be called whenever state changes
- * @param onUnreadChange a callback that will be called with the number of new unread messages
- */
-export function useHCWindowCommunicator(
-	onStateChange: ( state: WindowState ) => void,
-	onUnreadChange: ( unreadCount: number ) => void
-) {
-	const { selectedSite, subject, message, userDeclaredSite } = useSelect( ( select ) => {
+export function useHCWindowCommunicator( isMinimized: boolean ) {
+	const { selectedSite, subject, message, userDeclaredSite, iframe } = useSelect( ( select ) => {
 		return {
 			selectedSite: select( HELP_CENTER_STORE ).getSite(),
 			userDeclaredSite: select( HELP_CENTER_STORE ).getUserDeclaredSite(),
 			subject: select( HELP_CENTER_STORE ).getSubject(),
 			message: select( HELP_CENTER_STORE ).getMessage(),
+			iframe: select( HELP_CENTER_STORE ).getIframe(),
 		};
 	} );
-
+	const queryClient = useQueryClient();
+	const [ unreadCount, setUnreadCount ] = useState( 0 );
+	const [ chatStatus, setChatStatus ] = useState( '' );
 	const siteId = useSelector( getSelectedSiteId );
 	const currentSite = useSelect( ( select ) => select( SITE_STORE ).getSite( siteId ), [ siteId ] );
-
-	const queryClient = useQueryClient();
+	const chatWindow = iframe?.contentWindow;
+	function happyChatPostMessage( message: Record< string, string > ) {
+		chatWindow?.postMessage( message, '*' );
+	}
 
 	const supportSite = selectedSite || userDeclaredSite || currentSite;
 	const { resetStore } = useDispatch( HELP_CENTER_STORE );
 	useHappychatAuth();
 
+	useEffect( () => {
+		isMinimized
+			? happyChatPostMessage( { type: 'window-state-change', state: 'minimized' } )
+			: happyChatPostMessage( { type: 'window-state-change', state: 'maximized' } );
+	}, [ chatWindow, isMinimized ] );
+
+	const closeChat = () => {
+		setChatStatus( '' );
+		setUnreadCount( 0 );
+		happyChatPostMessage( { type: 'window-state-change', state: 'closed' } );
+	};
+
 	useEffect( () => {
 		const messageHandler = ( event: MessageEvent ) => {
-			// please be careful about changing this check
-			// sometimes we send sensitive information to this event's origin,
-			// and changing this to something untrusted is a serious security risk.
```

Comment:
+1

Reply:
This is probably the only comment that I could see adding value to the code.

Reply:
👍🏼  Let's bring it back.